nacl-fast.js

```diff
@@ -53,6 +53,28 @@ function crypto_verify_32(x, xi, y, yi) {
   return vn(x,xi,y,yi,32);
 }
 
+function key_arrays_to_keypair(pk, sk) {
+  var pkArray, skArray;
+  return Object.defineProperties({}, {
+    publicKey: {
+      get: function() {
+        if (pkArray === undefined) {
+          pkArray = Object.freeze(Array.from(pk));
+        }
+        return pkArray;
+      }
+    },
+    secretKey: {
+      get: function() {
+        if (skArray === undefined) {
+          skArray = Object.freeze(Array.from(sk));
+        }
+        return skArray;
+      }
+    }
+  });
+}
```

Comment:
This produces an object `{publicKey: [...], secretKey: [...]}` that can't be mutated.